allow formatting of git.commit.time

Nic128 opened the pull request:
I noticed Springboot `/info` route only reads from `git.properties`. It does not convert it further at runtime. You can modify the timestamp format when you generate `git.properties`, but after that, Springboot just reads it.

```
    <build>
        <finalName>yourprojectname</finalName>
        <plugins>
            <plugin>
                <groupId>pl.project13.maven</groupId>
                <artifactId>git-commit-id-plugin</artifactId>
                <version>3.0.1</version>
                <configuration>
                    <verbose>false</verbose>
                    <dateFormat>yyyy-MM-dd'T'HH:mm:ssZ</dateFormat>
                    <generateGitPropertiesFile>true</generateGitPropertiesFile>
                    <generateGitPropertiesFilename>${project.build.outputDirectory}/git.properties
                    </generateGitPropertiesFilename>
                    <excludeProperties>
                        <excludeProperty>git.remote.origin.url</excludeProperty>
                    </excludeProperties>
                </configuration>
            </plugin>
        </plugins>
    </build>
```

This pom xml setting gives an output like this from Springboot, I just verified:

```
{
  "git": {
    "commit": {
      "time": "2020-07-03T14:55:38Z",
      "id": "...."
    },
    "branch": "master"
  },
  "build": {
    ...
    "time": "2020-07-03T14:56:01.156Z"
  }
}
```

This PR is to make `express-actuator` match what Springboot does. By default, we should show `git.commit.time` as is printed in `git.properties`. And I am providing a new option to format at runtime, `infoDateFormat` (Using [moment format](https://momentjs.com/docs/#/displaying/format/)), if needed.

Using node-git-info, you already print to ISO string. https://github.com/rcruzper/node-git-info/blob/master/lib/gitCommands.js#L44

I am thinking of putting the date formatting capability on node-git-info directly and just have express-actuator read the result.

rcruzper commented:
Hi, great feature! thank you so much for contribute to this project :)

It seems like there is an issue with your changes, it seems related to timezone.
Also, do you think you could add a functional test to verify when an incorrect format is set as infoDateFormat and take into account this case?

Nic128 commented:
If I look at https://github.com/n0mer/gradle-git-properties, the date format option and timezone are handled at the git.properties creation. I can create the feature and tests in node-git-info.

Then simplify this PR to only read values.

rcruzper commented on test/functional/info.test.js:
```diff
@@ -117,7 +117,42 @@ describe('GET /info with infoGitMode', function() {
+                expect(res.body.build.version).to.equal("1.0.0");
+                expect(res.body.git.branch).to.equal('master');
+                expect(res.body.git.commit.id).to.equal('1a24c24');
+                expect(res.body.git.commit.time).to.equal('2016-11-18 08:16:39-0500');
```
I would change this in order to fix the test, due to the usage of UTC as timezone by default in the tests
```suggestion
                expect(res.body.git.commit.time).to.equal('2016-11-18 08:16:39+0000');
```

Nic128 replied:
It would either fail locally, or in Travis. To make it work everywhere for the test, I removed the timezone from the test date format.

rcruzper commented:
> If I look at https://github.com/n0mer/gradle-git-properties, the date format option and timezone are handled at the git.properties creation. I can create the feature and tests in node-git-info.
> 
> Then simplify this PR to only read values.

I think it's good to have the feature here, this way we can change the format without running node-git-info tool again

Nic128 commented:
I added another feature after fixing the tests. Since springboot can put any information in the build object, I introduced this.

`feature: allow adding information to the build object`

I wanted to add the build time in my usecase, I just need to pass it down to express-actuator.

![Screen Shot 2020-07-10 at 9 37 23 AM](https://user-images.githubusercontent.com/7115401/87160468-0188a400-c291-11ea-92ad-3498bf37f5b8.png)

rcruzper commented on test/functional/info.test.js:
```diff
@@ -117,9 +117,79 @@ describe('GET /info with infoGitMode', function() {
+                expect(res.body.build.version).to.equal("1.0.0");
+                expect(res.body.git.branch).to.equal('master');
+                expect(res.body.git.commit.id).to.equal('1a24c24');
+                expect(res.body.git.commit.time).to.equal('2016-11-18 08:16:39');
```
```suggestion
                expect(res.body.git.commit.time).to.equal('2016-11-18 13:16:39');
```

Nic128 replied:
Does travis execute automatically after a push? I should be able to see this failure.

rcruzper commented:
thanks for the contribution
